include/picongpu/fields/absorber/Absorber.hpp

```diff
@@ -178,6 +182,116 @@ namespace detail
         { Absorber::zNegativeNumCells, Absorber::zPositiveNumCells }
     };
 
+    //! Thickness of the absorbing layer
+    class Thickness
+    {
+    public:
+
+        //! Create a zero thickness
+        Thickness()
+        {
+            for( uint32_t axis = 0u; axis < 3u; axis++ )
+                for( uint32_t direction = 0u; direction < 2u; direction++ )
+                    (*this)( axis, direction ) = 0u;
+        }
+
+        /** Get thickness for the given boundary
+         *
+         * @param axis axis, 0 = x, 1 = y, 2 = z
+         * @param direction direction, 0 = negative (min coordinate),
+         *                  1 = positive (max coordinate)
+         */
+        uint32_t operator()(
+            uint32_t const axis,
+            uint32_t const direction
+        ) const
+        {
+            return numCells[ axis ][ direction ];
+        }
+
+        /** Get reference to thickness for the given boundary
+         *
+         * @param axis axis, 0 = x, 1 = y, 2 = z
+         * @param direction direction, 0 = negative (min coordinate),
+         *                  1 = positive (max coordinate)
+         */
+        uint32_t & operator()(
+            uint32_t const axis,
+            uint32_t const direction
+        )
+        {
+            return numCells[ axis ][ direction ];
+        }
+
+    private:
+
+        /** Number of absorber cells along each boundary
+         *
+         * First index: 0 = x, 1 = y, 2 = z.
+         * Second index: 0 = negative (min coordinate), 1 = positive (max coordinate).
+         */
+        uint32_t numCells[ 3 ][ 2 ];
+
+    };
+
+    /** Get absorber thickness in number of cells for the global domain
+     *
+     * This function takes into account which boundaries are periodic and
+     * absorbing.
+     */
+    inline Thickness getGlobalThickness()
+    {
+        Thickness thickness;
+        for( uint32_t axis = 0u; axis < 3u; axis++ )
+            for( uint32_t direction = 0u; direction < 2u; direction++ )
+                thickness( axis, direction ) = numCells[ axis ][ direction ];
+        const DataSpace< DIM3 > isPeriodicBoundary =
```

Comment:
You assign a N-DIM value which can be 2 to a 3 dimensional value. This should fail.

Reply:
Discussed offline, the value on the right side is also always 3d. In case of a 2d simulation, the z axis is always set to periodic (which makes sense).